lua/neogit/lib/git/rebase.lua

```diff
@@ -92,6 +92,17 @@ function M.reword(commit, message)
   fire_rebase_event("ok")
 end
 
+function M.modify(commit)
+  local short_commit = string.sub(commit, 1, 7)
+  local editor = "sed -i 's/^pick " .. short_commit .. "/edit " .. short_commit .. "/'"
```

Comment:
Couple notes here. Something I learned the hard way is that a 7 character hash isn't a constant thing. Turns out (https://github.com/NeogitOrg/neogit/pull/1046), for every order-of-magnitude of commits in a repo, git automatically adds another character to the hash length! So, we can't count on it being 7.

Reply:
Also, tried this out, and I get this error:

```
git rebase -i --autosquash --autostash 4c29337d72cbdb8187965e6839c18274d09508ea^:

    sed: 1: "/Users/cameron/code/neo ...": command c expects \ followed by text
    error: There was a problem with the editor 'sed -i 's/^pick 4c29337/edit 4c29337/''.
```

Reply:
Hmm. I agree it would be nice to only depend on neovim. This works for me in the shell:
`GIT_SEQUENCE_EDITOR="nvim` -c '%s/^pick \(<SHORT_SHA>.*\)/edit \1/' -c 'wq'" git rebase -i  <LONG_SHA>^`
but not when I do:
` local editor="nvim -c '%s/^pick \\(" .. short_commit .. ".*\\)/edit \\1/' -c 'wq'"`.

The downside of matching like this in a huge repo is that there could be multiple matches on the 7 character hash which would edit two commits... not sure if that's acceptable or how to work around that.

Reply:
I added `in_pty(true)` to the call and then it worked.

Reply:
Does it work with the full sha?

Reply:
We could make a helper that calls `git rev-list --count --all` to determine the correct length of the hash. It would be handy in some other places to, so the ui layer shows the correct length of hash

Reply:
The command you sent just gives me some large number, e.g. in this repo 2973... however `git log -n 1 --format=%h` returns a hash where we can count the characters. I added a helper function in lib.git.log.

This repo has 8 characters in the hash and that works fine for me now.

Reply:
I made a similar implementation for "drop a commit". Should I add that in a separate pr or can I add it here?

Reply:
Awesome - let's just do that in a new PR :)

Reply:
Lets just do it in a new PR - sounds great :D